### PR TITLE
Add ld_preload to images

### PR DESCRIPTION
**Pull request by mklotton**

### CODE REVIEW

**Comment by mklotton** [comment]
#32

File: docker/ubuntu-based/dev/Dockerfile [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
 # optional argument - a specific VPP commit ID
 ARG VPP_COMMIT_ID="xxx"
 
 # set work directory
 WORKDIR /root/
 
+ENV VPP_DIR /opt/vpp-agent/dev/vpp
+ENV LDP_DIR $VPP_DIR/extras
+ENV VCL_LDPRELOAD_LIB_DIR $LDP_DIR/vcl-ldpreload/src/.libs
+ENV LD_PRELOAD_INSTALL /opt/ldpreload
```
Comment (rastislavszabo) [inline docker/ubuntu-based/dev/Dockerfile]: LD_PRELOAD_INSTALL_DIR

(all others have _DIR suffix)

File: docker/ubuntu-based/dev/Dockerfile [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
-FROM ligato/dev-vpp-agent:v1.0.5
+FROM ligato/dev-vpp-agent:v1.0.5-4-gd26bfb8
 
 # optional argument - a specific VPP commit ID
 ARG VPP_COMMIT_ID="xxx"
 
 # set work directory
 WORKDIR /root/
 
+ENV VPP_DIR /opt/vpp-agent/dev/vpp
```
Comment (rastislavszabo) [inline docker/ubuntu-based/dev/Dockerfile]: add come comment above..

File: docker/ubuntu-based/dev/Dockerfile [hunk 3]
```diff
@@ -1,27 +1,50 @@
...
 
+# build vcl-ldpreload and uri dependency
+RUN /bin/bash -c " \
+    cd ${VPP_DIR} && \
+    perl -pi -e 's/noinst_PROGRAMS/bin_PROGRAMS/g' $VPP_DIR/src/uri.am && \
+    yes | make install-dep && \
+    make bootstrap && \
+    make pkg-deb && \
+    cd build-root && \
+    dpkg -i vpp_*.deb vpp-dev_*.deb vpp-lib_*.deb vpp-plugins_*.deb && \
```
Comment (rastislavszabo) [inline docker/ubuntu-based/dev/Dockerfile]: do we need to run make pkg-deb and dpkg -i always?

File: docker/ubuntu-based/dev/Dockerfile [hunk 4]
```diff
@@ -1,27 +1,50 @@
...
+    cd $LDP_DIR/vcl-ldpreload/src && \
+    autoreconf -i -f && \
+    ./configure VPP_DIR=$VPP_DIR && \
+    make"
+
+# copy vcl-ldpreload
```
Comment (rastislavszabo) [inline docker/ubuntu-based/dev/Dockerfile]: only vcl-ldpreload? copy to host?

File: k8s/contiv-vpp.yaml [hunk 5]
```diff
@@ -96,6 +96,29 @@ spec:
       hostNetwork: true
       hostPID: true
 
+      initContainers:
```
Comment (rastislavszabo) [inline k8s/contiv-vpp.yaml]: add some comment above, explaining what this does

**Pull request merged**